libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1236,22 +1236,39 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 		case FunctionType::Kind::ABIEncode:
 		case FunctionType::Kind::ABIEncodePacked:
 		case FunctionType::Kind::ABIEncodeWithSelector:
+		case FunctionType::Kind::ABIEncodeCall:
 		case FunctionType::Kind::ABIEncodeWithSignature:
 		{
 			bool const isPacked = function.kind() == FunctionType::Kind::ABIEncodePacked;
 			bool const hasSelectorOrSignature =
 				function.kind() == FunctionType::Kind::ABIEncodeWithSelector ||
+				function.kind() == FunctionType::Kind::ABIEncodeCall ||
 				function.kind() == FunctionType::Kind::ABIEncodeWithSignature;
 
 			TypePointers argumentTypes;
-			TypePointers targetTypes;
-			for (unsigned i = 0; i < arguments.size(); ++i)
+
+			ASTNode::listAccept(arguments, *this);
+
+			if (function.kind() == FunctionType::Kind::ABIEncodeCall)
 			{
-				arguments[i]->accept(*this);
-				// Do not keep the selector as part of the ABI encoded args
-				if (!hasSelectorOrSignature || i > 0)
-					argumentTypes.push_back(arguments[i]->annotation().type);
+				solAssert(arguments.size() == 2, "");
+
+				auto const tupleType = dynamic_cast<TupleType const*>(arguments[1]->annotation().type);
+
+				// Account for tuples with one component which become that component
+				if (!tupleType)
+					argumentTypes.emplace_back(arguments[1]->annotation().type);
+				else
+					argumentTypes = tupleType->components();
```

Comment:
```suggestion
				// Account for tuples with one component which become that component
				if (auto const* tupleType = dynamic_cast<TupleType const*>(arguments[1]->annotation().type))
					argumentTypes = tupleType->components();
				else
					argumentTypes.emplace_back(arguments[1]->annotation().type);
```

Reply:
I am marking this as resolved as the updates made it obsolete